src/test/java/htsjdk/samtools/cram/build/ContainerFactoryTest.java

```diff
@@ -21,105 +19,499 @@
  */
 public class ContainerFactoryTest extends HtsjdkTest {
 
-    @Test
-    public void testUnmapped() {
-        SAMFileHeader header = new SAMFileHeader();
+    public static Container getSingleRefContainer(final SAMFileHeader samFileHeader) {
+        final ContainerFactory factory = new ContainerFactory(samFileHeader, 10);
+        final List<CramCompressionRecord> records = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            final CramCompressionRecord record = new CramCompressionRecord();
+            record.readBases = "AAA".getBytes();
+            record.qualityScores = "!!!".getBytes();
+            record.readLength = 3;
+            record.readName = "" + i;
+            record.sequenceId = 0;
+            record.alignmentStart = i + 1;
+            record.setLastSegment(true);
+            record.readFeatures = Collections.emptyList();
 
-        int recordsPerContainer = 10;
-        ContainerFactory factory = new ContainerFactory(header, recordsPerContainer);
+            if (i % 2 == 0) {
+                record.setSegmentUnmapped(true);
+            } else {
+                record.setSegmentUnmapped(false);
+            }
+
+            records.add(record);
+        }
+
+        final Container container = factory.buildContainer(records);
+        Assert.assertEquals(container.nofRecords, 10);
+        Assert.assertEquals(container.sequenceId, 0);
+        Assert.assertEquals(container.slices.length, 1);
+        Assert.assertEquals(container.slices[0].sequenceId, container.sequenceId);
+        return container;
+    }
 
-        List<CramCompressionRecord> records = new ArrayList<>();
-        for (int i = 0; i < recordsPerContainer; i++) {
+    public static Container getUnmappedNoRefContainer(final SAMFileHeader samFileHeader) {
```

Comment:
There are two ways to be unmapped/unplaced.  No Ref ID is one, no Alignment Start is the other